flink-core/src/main/java/org/apache/flink/util/CompressedSerializedValue.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+package org.apache.flink.util;
+
+import java.io.IOException;
+
+/** An extension of {@link SerializedValue} that compresses the value after the serialization. */
+public class CompressedSerializedValue<T> extends SerializedValue<T> {
+
+    private static final long serialVersionUID = -4358765382738374654L;
+
+    private CompressedSerializedValue(byte[] serializedData) {
```

Comment:
looks to me this constructor and the `fromBytes()` method are not used in following commits.

Reply:
I think there is no need to add a production method which is only used by its own unit test.

Reply:
> I think there is no need to add a production method which is only used by its own unit test.

Thank you for pointing this out. I'll take care of this the next time. Resolved.

Reply:
I think I was wrong that this method and `fromBytes()` method are needed when loading and deserialize blobs.
I think we need java docs for it and rename the param as `compressedSerializedData`.

Reply:
`CompressedSerializedValue#frombytes` has been moved to the second pull request.